### ISSUE TITLE
Removal of `Cargo.lock` from `rust-src` causes Bevy CI to fail

### ISSUE THREAD

**Issue by BD103** [rust-lang/rust#128840]
## Description

- Bevy issue: https://github.com/bevyengine/bevy/issues/14662
- Regression PR: #128685
- Original CI job: [`ci.yml`](https://github.com/bevyengine/bevy/blob/8c2e70b7446ecab3c08f62e288de294f605c6100/.github/workflows/ci.yml#L168-L176)

As part of Bevy's CI, we check that it builds with `wasm32-unknown-unknown` with `target-feature=+atomics,+bulk-memory` on nightly. This check has recently started failing due to #128685. I found this after running `cargo-bisect-rustc`, whose output can be found below:

## `cargo-bisect-rustc` output

searched nightlies: from nightly-2024-08-06 to nightly-2024-08-08
regressed nightly: nightly-2024-08-08
searched commit range: https://github.com/rust-lang/rust/compare/60d146580c10036ce89e019422c6bc2fd9729b65...8b3870784f69d8ed2ae4246d1b56f150e3131499
regressed commit: https://github.com/rust-lang/rust/commit/0ddead30f02a855b794bb31b7356f228bd30aa99

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.9</summary>

Host triple: aarch64-apple-darwin
Reproduce with:
```bash
RUSTFLAGS="-C target-feature=+atomics,+bulk-memory" GITHUB_TOKEN=`gh auth token` cargo bisect-rustc --access=github --with-src --start=2024-08-06 --end=2024-08-08 -- check --target wasm32-unknown-unknown -Z build-std=std,panic_abort
```
</details>

**Comment by ChrisDenton** [comment]
This is a Cargo issue and was fixed by https://github.com/rust-lang/cargo/pull/14370

**Comment by BD103** [comment]
> This is a Cargo issue and was fixed by [rust-lang/cargo#14370](https://github.com/rust-lang/cargo/pull/14370)

Wonderful! I'll close this and test it tomorrow!
